File: vault/android/app/src/main/java/com/vault/VonageVerifySilentAuthModule.kt
package com.vault

import android.util.Log
import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import com.vonage.numberverification.VGNumberVerificationClient
import com.vonage.numberverification.VGNumberVerificationParameters
import org.json.JSONObject
import java.net.URL

class VonageVerifySilentAuthModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {
    val client = VGNumberVerificationClient.initializeSdk(reactContext)
    override fun getName() = "VonageVerifySilentAuthModule"


    @ReactMethod
    fun openWithDataCellular(url: String, debug: Boolean, promise: Promise) {
        Log.d("openWithDataCellular", "openWithDataCellular url: $url, debug: $debug")
        val params = VGNumberVerificationParameters(
          url = url,
          headers = mapOf("x-my-header" to "My Value") ,
          queryParameters = mapOf("query-param" to "value"),
          maxRedirectCount = 20
        )

        val resp: JSONObject =  VGNumberVerificationClient.getInstance().startNumberVerification(params, debug)

        if (resp.optString("error") != "") {
            promise.reject("openWithDataCellular failed: ", resp.optString("error"))
        } else {
            val status = resp.optInt("http_status")
            if (status == 200) {
                promise.resolve(resp.toString())
            } else {
                promise.reject("openWithDataCellular failed: ", status.toString())
            }
        }
    }
}

File: vault/ios/VonageVerifySilentAuthModule.swift
//
//  VonageVerifySilentAuthModule.swift
//  VonageVault
//
//  Created by iujie on 20/02/2025.
//

import Foundation
import VonageClientSDKNumberVerification


@objc(VonageVerifySilentAuthModule)
class VonageVerifySilentAuthModule: NSObject {
  let client = VGNumberVerificationClient()

  @objc static func requiresMainQueueSetup() -> Bool { return true }
  
  @objc public func openWithDataCellular(
    _ url: String,
    debug: Bool,
     resolver resolve: @escaping RCTPromiseResolveBlock,
     rejecter reject: @escaping RCTPromiseRejectBlock
  ) -> Void {
     let params = VGNumberVerificationParameters(url: url,
                                                 headers: ["x-my-header": "My Value"],
                                                 queryParameters: ["query-param" : "value"],
                                                 maxRedirectCount: 20)
    Task {
      do {
        let response = try await client.startNumberVerification(params: params, debug: debug)
        let status = response["http_status"] as? Int
        if (status == 200) {
          resolve("\(jsonToString(json: response as AnyObject))");
          
        } else {
          reject("Failed openWithDataCellular", "\(response["error_description"] ?? jsonToString(json: response as AnyObject))", "\(String(describing: status))" as? Error)
        }
      }
      catch {
        reject("Failed openWithDataCellular", error.localizedDescription, error)
      }
    }
   }
  
  func jsonToString(json: AnyObject) -> String{
      do {
          let data1 = try JSONSerialization.data(withJSONObject: json, options: JSONSerialization.WritingOptions.prettyPrinted)
          let convertedString = String(data: data1, encoding: String.Encoding.utf8) as NSString? ?? ""
          debugPrint(convertedString)
          return convertedString as String
      } catch let myJSONError {
          debugPrint(myJSONError)
          return ""
      }
  }
}

